chiselFrontend/src/main/scala/Chisel/When.scala

```diff
@@ -2,9 +2,12 @@
 
 package Chisel
 
+import scala.language.experimental.macros
+
 import internal._
 import internal.Builder.pushCommand
 import internal.firrtl._
+import internal.sourceinfo.{SourceInfo, WhenTransform}
 
 object when {  // scalastyle:ignore object.name
```

Comment:
Using macros to transform API calls is still complicated even though this is likely the simplest way to achieve the desired result. The 'over'complicated parts were calls that were transformed which didn't need to be transformed.

Similarly, my goal in restricting macro transformations to only functions that need it is so that future developers adding to the chisel API do not blindly add macro transformations for their calls (unless they actually need them). This is basically protection against the SWE phenomena that familiarity with code declines proportionally to time since commit.